### DISCUSSION TITLE
Subpixel smoothing disabled not working properly?

### DISCUSSION THREAD

**Discussion by manujmg** [NanoComp/meep#2756] [Q&A] [answered]
Hello everyone,

I am running a simulation and don't want to have the subpixel smoothing activated because it creates a weird effect around the edges of my metals. 

I am attaching two plots of epsilon data in one specific y-slice of my 3D simulation. 

The following image is with eps_averaging = True
<img width="571" alt="Captura de pantalla 2024-01-08 a las 18 17 41" src="https://github.com/NanoComp/meep/assets/26410461/76fae863-bed2-453e-94f4-7b42594d4bdc">

The next picture is with eps_averaging = False and the result is almost identical to the above picture except for some portions of my dielectric structure (red circles). Do you know why the edges of the metals do not change?
<img width="571" alt="Captura de pantalla 2024-01-08 a las 18 23 59" src="https://github.com/NanoComp/meep/assets/26410461/e82de004-e2af-4563-b973-5de6ff8ac2cf">

**Comment by smartalecH** [comment]
A couple things to note:

First, meep doesn't do any subpixel smoothing in voxels containing metals. That's too hard (unfortunately). So you shouldn't expect a difference in these regions regardless of what the flag is set to. 

Second, those interfaces "look" weird because what meep is "visualizing" as the permittivity is not what you are actually simulating (apples to apples). The permittivity is a tensor, and different components lie on different points of the yee cell. Same with the conductivity. So meep takes the trace of the the tensor within a cell, and plots that. It gets a little hairy when you have dispersive materials/metals (check out the docs on supplying a frequency if you don't have a source specified).

**Reply by manujmg** [reply]
So... Basically meep simulation is working completely fine but the permittivity output is weird because of the dispersive materials so it should not be used as a reference for plotting?

**Reply by smartalecH** [reply, marked as answer]
Yep. Visualizing this kind of stuff is tricky thanks to both the Yee grid and the frequency-dependent nature of materials. So you do what you can.
